instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/patch.py

```diff
@@ -0,0 +1,133 @@
+import redis
+from wrapt import ObjectProxy, wrap_function_wrapper
+
+from opentelemetry import trace
+
+from .util import _extract_conn_attributes, format_command_args
+from .version import __version__
+
+_DEFAULT_SERVICE = "redis"
+_RAWCMD = "redis.raw_command"
+
+
+def patch():
+    """Patch the instrumented methods
+
+    This duplicated doesn't look nice. The nicer alternative is to use an ObjectProxy on top
+    of Redis and StrictRedis. However, it means that any "import redis.Redis" won't be instrumented.
+    """
+    if getattr(redis, "_opentelemetry_patch", False):
+        return
+    setattr(redis, "_opentelemetry_patch", True)
+
+    if redis.VERSION < (3, 0, 0):
+        wrap_function_wrapper(
+            "redis", "StrictRedis.execute_command", traced_execute_command
+        )
+        wrap_function_wrapper("redis", "StrictRedis.pipeline", traced_pipeline)
+        wrap_function_wrapper("redis", "Redis.pipeline", traced_pipeline)
+        wrap_function_wrapper(
+            "redis.client", "BasePipeline.execute", traced_execute_pipeline
+        )
+        wrap_function_wrapper(
+            "redis.client",
+            "BasePipeline.immediate_execute_command",
+            traced_execute_command,
+        )
+    else:
+        wrap_function_wrapper(
+            "redis", "Redis.execute_command", traced_execute_command
+        )
+        wrap_function_wrapper("redis", "Redis.pipeline", traced_pipeline)
+        wrap_function_wrapper(
+            "redis.client", "Pipeline.execute", traced_execute_pipeline
+        )
+        wrap_function_wrapper(
+            "redis.client",
+            "Pipeline.immediate_execute_command",
+            traced_execute_command,
+        )
+
+
+def unwrap(obj, attr):
+    func = getattr(obj, attr, None)
+    if isinstance(func, ObjectProxy) and hasattr(func, "__wrapped__"):
+        setattr(obj, attr, func.__wrapped__)
+
+
+def unpatch():
+    if getattr(redis, "_opentelemetry_patch", False):
+        setattr(redis, "_opentelemetry_patch", False)
+        if redis.VERSION < (3, 0, 0):
+            unwrap(redis.StrictRedis, "execute_command")
+            unwrap(redis.StrictRedis, "pipeline")
+            unwrap(redis.Redis, "pipeline")
+            unwrap(
+                redis.client.BasePipeline,  # pylint:disable=no-member
+                "execute",
+            )
+            unwrap(
+                redis.client.BasePipeline,  # pylint:disable=no-member
+                "immediate_execute_command",
+            )
+        else:
+            unwrap(redis.Redis, "execute_command")
+            unwrap(redis.Redis, "pipeline")
+            unwrap(redis.client.Pipeline, "execute")
+            unwrap(redis.client.Pipeline, "immediate_execute_command")
+
+
+#
+# tracing functions
+#
+def traced_execute_command(func, instance, args, kwargs):
+    tracer = trace.get_tracer(_DEFAULT_SERVICE, __version__)
+    query = format_command_args(args)
+    with tracer.start_as_current_span(query) as span:
```

Comment:
From looking at the output from the original DD code, that was what was being emitted to as a name for the span, happy to change this back to the command name though.

Reply:
Reverted to using "redis.command" which was a previous identifier in the dd code. might be more useful in the future to update this to "redis.*" to match the command that was used ie. redis.hget, without all the parameters. this would ensure a pretty low cardinality